src/Builder/ClassMetadataBuilder/ClassMetadataBuilder.php

```diff
@@ -151,10 +151,38 @@ protected function addSourceAndTarget(array $manifest, ClassMetadataTransfer $tr
      */
     protected function addPositions(array $manifest, ClassMetadataTransfer $transfer): ClassMetadataTransfer
     {
-        $transfer->setBefore(ltrim($manifest[IntegratorConfig::MANIFEST_KEY_POSITION][IntegratorConfig::MANIFEST_KEY_POSITION_BEFORE] ?? '', '\\'));
-        $transfer->setAfter(ltrim($manifest[IntegratorConfig::MANIFEST_KEY_POSITION][IntegratorConfig::MANIFEST_KEY_POSITION_AFTER] ?? '', '\\'));
+        $transfer->setBefore($this->getPositions($manifest, IntegratorConfig::MANIFEST_KEY_POSITION_BEFORE));
+        $transfer->setAfter($this->getPositions($manifest, IntegratorConfig::MANIFEST_KEY_POSITION_AFTER));
         $transfer->setCondition($manifest[IntegratorConfig::MANIFEST_KEY_CONDITION] ?? null);
 
         return $transfer;
     }
+
+    /**
+     * @param array $manifest
+     * @param string $positionKey
+     *
+     * @return \ArrayObject
+     */
+    protected function getPositions(array $manifest, string $positionKey): ArrayObject
+    {
+        $positions = new ArrayObject();
+
+        $positionData = $manifest[IntegratorConfig::MANIFEST_KEY_POSITION][$positionKey] ?? null;
+        if (!$positionData) {
+            return $positions;
+        }
+
+        if (is_string($positionData)) {
+            $positions->append(ltrim($positionData, '\\'));
+
+            return $positions;
+        }
+
+        foreach ($positionData as $position) {
+            $positions->append(ltrim($position, '\\'));
```

Comment:
In general case it's better to extract the repeatable functionality into a new method. I mean `ltrim($position, '\\')`

Reply:
Ok, created removeSlashFromFromBeginningOfString method

Reply:
You used too much "From" in the name of the method :) Anyway I think you can give it a shorter name, something like "removeLeadingSlash"

Reply:
Thanks ++